docs/index.rst

```diff
@@ -1,25 +1,379 @@
+
+- Verified HTTPS connections
+- Uploading doesn't require executing ``setup.py``
+- Uploading files that have already been created, allowing testing of
+    distributions before release
+- Supports uploading any packaging format (including `wheels`_)
+
+Installation
+------------
+
+.. code-block:: console
+
+    $ pip install twine
+
+Using Twine
+-----------
+
+1. Create some distributions in the normal way:
+
+    .. code-block:: console
+
+       $ python setup.py sdist bdist_wheel
+
+2. Upload with ``twine`` to `Test PyPI`_ and verify things look right.
+    Twine will automatically prompt for your username and password:
+
+    .. code-block:: console
+
+       $ twine upload -r testpypi dist/*
+       username: ...
+       password:
+       ...
+
+3. Upload to `PyPI`_:
+
+    .. code-block:: console
+
+       $ twine upload dist/*
+
+4. Done!
+
+More documentation on using Twine to upload packages to PyPI is in
+the `Python Packaging User Guide`_.
+
+Keyring Support
+---------------
+
+Instead of typing in your password every time you upload a distribution, Twine
+allows storing a username and password securely using `keyring`_.
+Keyring is installed with Twine but for some systems (Linux mainly) may
+require `additional installation steps`_.
+
+Once Twine is installed, use the ``keyring`` program to set a
+username and password to use for each package index (repository) to
+which you may upload.
+
+For example, to set a username and password for PyPI:
+
+.. code-block:: console
+
+    $ keyring set https://upload.pypi.org/legacy/ your-username
+
+or
+
+.. code-block:: console
+
+    $ python3 -m keyring set https://upload.pypi.org/legacy/ your-username
+
+and enter the password when prompted.
+
+For a different repository, replace the URL with the relevant repository
+URL. For example, for Test PyPI, use ``https://test.pypi.org/legacy/``.
+
+The next time you run ``twine``, it will prompt you for a username and will
+grab the appropriate password from the keyring.
+
+**Note:** If you are using Linux in a headless environment (such as on a
+server) you'll need to do some additional steps to ensure that Keyring can
+store secrets securely. See `Using Keyring on headless systems`_.
+
+Disabling Keyring
+^^^^^^^^^^^^^^^^^
+
+In most cases, simply not setting a password with ``keyring`` will allow Twine
+to fall back to prompting for a password. In some cases, the presence of
+Keyring will cause unexpected or undesirable prompts from the backing system.
+In these cases, it may be desirable to disable Keyring altogether. To disable
+Keyring, simply invoke:
+
+.. code-block:: console
+
+    $ keyring --disable
+
+or
+
+.. code-block:: console
+
+    $ python -m keyring --disable
+
+That command will configure for the current user the "null" keyring,
+effectively disabling the functionality, and allowing Twine to prompt
+for passwords.
+
+See `twine 338 <https://github.com/pypa/twine/issues/338>`_ for
+discussion and background.
+
+Options
```

Comment:
Non-blocking nit: I just noticed that this seems this is either describing sub-command options or the subcommands or both so `Options` is a less than idea section title.